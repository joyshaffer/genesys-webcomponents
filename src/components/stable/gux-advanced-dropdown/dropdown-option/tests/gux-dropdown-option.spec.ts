import { newSpecPage } from '@stencil/core/testing';
import { GuxDropdownOption } from '../gux-dropdown-option';

const components = [GuxDropdownOption];
const html = `<gux-dropdown-option text="TestsAreAwesome"></gux-dropdown-option>`;
const language = 'en';

describe('gux-dropdown-option', () => {
  it('should build', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.root).toMatchSnapshot();
  });

  describe('Class Logic', () => {
    describe('shouldFilter', () => {
      it('should not filter if no search string is provided', async () => {
        const page = await newSpecPage({ components, html, language });
        const component = page.root;
        const result = await component.shouldFilter('');

        await page.waitForChanges();

        expect(result).toBeFalsy();
        expect(page.root).toMatchSnapshot();
      });

      it('should not filter if string is in the text', async () => {
        const page = await newSpecPage({ components, html, language });
        const component = page.root;
        const result = await component.shouldFilter('Are');

        await page.waitForChanges();

        expect(result).toBeFalsy();
        expect(page.root).toMatchSnapshot();
      });

      it('should not filter if case-insensitive string is in the text', async () => {
        const page = await newSpecPage({ components, html, language });
        const component = page.root;
        const result = await component.shouldFilter('are');

        await page.waitForChanges();

        expect(result).toBeFalsy();
        expect(page.root).toMatchSnapshot();
      });

      it('should filter if string is not in the text', async () => {
        const page = await newSpecPage({ components, html, language });
        const component = page.root;
        const result = await component.shouldFilter('Not');

        await page.waitForChanges();

        expect(result).toBeTruthy();
        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
