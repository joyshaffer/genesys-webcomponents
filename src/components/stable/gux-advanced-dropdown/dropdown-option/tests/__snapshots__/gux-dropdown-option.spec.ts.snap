// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-dropdown-option Class Logic shouldFilter should filter if string is not in the text 1`] = `
<gux-dropdown-option tabindex="0" text="TestsAreAwesome">
  <div class="gux-dropdown-option" title="TestsAreAwesome">
    <span>
      TestsAreAwesome
    </span>
  </div>
</gux-dropdown-option>
`;

exports[`gux-dropdown-option Class Logic shouldFilter should not filter if case-insensitive string is in the text 1`] = `
<gux-dropdown-option tabindex="0" text="TestsAreAwesome">
  <div class="gux-dropdown-option" title="TestsAreAwesome">
    <span>
      Tests
      <strong>
        Are
      </strong>
      Awesome
    </span>
  </div>
</gux-dropdown-option>
`;

exports[`gux-dropdown-option Class Logic shouldFilter should not filter if no search string is provided 1`] = `
<gux-dropdown-option tabindex="0" text="TestsAreAwesome">
  <div class="gux-dropdown-option" title="TestsAreAwesome">
    <span>
      TestsAreAwesome
    </span>
  </div>
</gux-dropdown-option>
`;

exports[`gux-dropdown-option Class Logic shouldFilter should not filter if string is in the text 1`] = `
<gux-dropdown-option tabindex="0" text="TestsAreAwesome">
  <div class="gux-dropdown-option" title="TestsAreAwesome">
    <span>
      Tests
      <strong>
        Are
      </strong>
      Awesome
    </span>
  </div>
</gux-dropdown-option>
`;

exports[`gux-dropdown-option should build 1`] = `
<gux-dropdown-option tabindex="0" text="TestsAreAwesome">
  <div class="gux-dropdown-option" title="TestsAreAwesome">
    <span>
      TestsAreAwesome
    </span>
  </div>
</gux-dropdown-option>
`;
